src/main/java/org/symphonyoss/symphony/messageml/elements/Dialog.java

```diff
@@ -0,0 +1,146 @@
+package org.symphonyoss.symphony.messageml.elements;
+
+import static org.apache.commons.lang3.StringUtils.containsWhitespace;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+
+import org.symphonyoss.symphony.messageml.MessageMLContext;
+import org.symphonyoss.symphony.messageml.MessageMLParser;
+import org.symphonyoss.symphony.messageml.exceptions.InvalidInputException;
+import org.symphonyoss.symphony.messageml.markdown.nodes.form.DialogNode;
+import org.symphonyoss.symphony.messageml.util.ShortID;
+import org.symphonyoss.symphony.messageml.util.XmlPrintStream;
+import org.w3c.dom.Node;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class represents the Symphony Element Dialog which is represented with tag name "dialog".
+ * The messageML representation of the dialog element can contain the following attributes:
+ * <ul>
+ *   <li> id (required) -> used to identify the dialog </li>
+ *   <li> width -> used to specify dialog width, must be among values "small", "medium", "large", "full-width". Default value is medium</li>
+ *   <li> state -> used to specify dialog state, must be among values "open", "close". Default value is "false"</li>
+ * </ul>
+ * It can contain the following child tags:
+ * <ul>
+ *   <li> title (required) -> used to specify dialog title, of type {@link DialogChild.Title}</li>
+ *   <li> body (required) -> used to specify dialog body, of type {@link DialogChild.Body}</li>
+ *   <li> footer -> used to specify dialog footer, of type {@link DialogChild.Footer}</li>
+ * </ul>
+ */
+public class Dialog extends Element {
+
+  public static final String MESSAGEML_TAG = "dialog";
+  public static final int ID_MAX_LENGTH = 64;
+
+  public static final String STATE_ATTR = "state";
+  public static final String FALSE_STATE = "false";
+  public static final List<String> ALLOWED_STATE_VALUES = Arrays.asList(FALSE_STATE, "open", "close");
```

Comment:
It should be only "open" or "close".
(My bad there are 'open', 'close', 'false' in the spec, I made a mistake sorry)